pkg/operator/configobservation/images/observe_images.go

```diff
@@ -122,9 +126,11 @@ func ObserveAllowedRegistriesForImport(genericListers configobserver.Listers, ex
 	if err != nil {
 		return prevObservedConfig, append(errs, err)
 	}
-	err = unstructured.SetNestedSlice(prevObservedConfig, o, allowedRegistriesForImportPath...)
-	if err != nil {
-		return prevObservedConfig, append(errs, err)
+	if len(o) > 0 {
```

Comment:
same here, we need to be able to blank out this value if all the allowedregistries are removed.